Docs for experimental endpoints (#1887)
stellar/go: services/horizon/internal/docs/reference/endpoints/path-finding-strict-receive.md

---
title: Find Payment Paths (Strict Receive)
clientData:
  laboratoryUrl: https://www.stellar.org/laboratory/#explorer?resource=paths&endpoint=all
---

The Stellar Network allows payments to be made across assets through _path payments_.  A path
payment specifies a series of assets to route a payment through, from source asset (the asset
debited from the payer) to destination asset (the asset credited to the payee).

A [Path Payment Strict Receive](../../../guides/concepts/list-of-operations.html#path-payment-strict-receive) allows a user to specify the *amount of the asset received*. The amount sent varies based on offers in the order books.  If you would like to search for a path specifying the amount to be sent, use the [Find Payment Paths (Strict Send)](./path-finding-strict-send.html).

A strict receive path search is specified using:

- The source account id or source assets.
- The asset and amount that the destination account should receive.

As part of the search, horizon will load a list of assets available to the source account id and
will find any payment paths from those source assets to the desired destination asset. The search's

> amount parameter will be used to determine if there a given path can satisfy a payment of the

Reviewer: ```suggestion
amount parameter will be used to determine if a given path can satisfy a payment of the
```

desired amount.

## Request

```
GET /paths/strict-receive?source_account={sa}&destination_asset_type={at}&destination_asset_code={ac}&destination_asset_issuer={di}&destination_amount={amount}&destination_account={da}
```

## Arguments

| name | notes | description | example |
| ---- | ----- | ----------- | ------- |

> | `?source_account` | string | The sender's account id. Any returned path must use a source that the sender can hold | `GARSFJNXJIHO6ULUBK3DBYKVSIZE7SC72S5DYBCHU7DKL22UXKVD7MXP` |

Reviewer: What does "a source that the sender can hold" mean?  Should that be "an asset that the sender can hold"?

> | `?source_account` | string | The sender's account id. Any returned path must use a source that the sender can hold | `GARSFJNXJIHO6ULUBK3DBYKVSIZE7SC72S5DYBCHU7DKL22UXKVD7MXP` |

Reviewer: Same as before: should the fields in this table start with a `?`

> | `?source_assets` | string | A comma separated list of assets. Any returned path must use a source included in this list  | `USD:GAEDTJ4PPEFVW5XV2S7LUXBEHNQMX5Q2GM562RJGOQG7GVCE5H3HIB4V,native` |

Reviewer: should "must use a source included in this list" be "must use an asset included in this list"?

| `?destination_account` | string | The destination account that any returned path should use | `GAEDTJ4PPEFVW5XV2S7LUXBEHNQMX5Q2GM562RJGOQG7GVCE5H3HIB4V` |
| `?destination_asset_type` | string | The type of the destination asset | `credit_alphanum4` |
| `?destination_asset_code` | string | The destination asset code, if destination_asset_type is not "native" | `USD` |

> | `?destination_asset_issuer` | string | The issuer for the destination, if destination_asset_type is not "native" | `GAEDTJ4PPEFVW5XV2S7LUXBEHNQMX5Q2GM562RJGOQG7GVCE5H3HIB4V` |

Reviewer: ```suggestion
| `?destination_asset_issuer` | string | The issuer for the destination asset, if destination_asset_type is not "native" | `GAEDTJ4PPEFVW5XV2S7LUXBEHNQMX5Q2GM562RJGOQG7GVCE5H3HIB4V` |
```

| `?destination_amount` | string | The amount, denominated in the destination asset, that any returned path should be able to satisfy | `10.1` |

The endpoint will not allow requests which provide both a `source_account` and a `source_assets` parameter. All requests must provide one or the other.
The assets in `source_assets` are expected to be encoded using the following format:

> The native asset should be represented as `"native"`. Issued assets should be represented as `"Code:IssuerAccountID"`. `"Code"` must consist of alphanumeric ASCII characters.

Reviewer: ```suggestion
XLM should be represented as `"native"`. Issued assets should be represented as `"Code:IssuerAccountID"`. `"Code"` must consist of alphanumeric ASCII characters.
```
